x-pack/metricbeat/module/googlecloud/_meta/docs.asciidoc

```diff
@@ -0,0 +1,55 @@
+
+```json
+{
+  "type": "service_account",
+  "project_id": "your-project-id",
+  "private_key_id": "a_private_key_id",
+  "private_key": "-----BEGIN PRIVATE KEY-----your private key\n-----END PRIVATE KEY-----\n",
+  "client_email": "some-email@your-project-id.iam.gserviceaccount.com",
+  "client_id": "123456",
+  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+  "token_uri": "https://oauth2.googleapis.com/token",
+  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/metricbeat-testing%40your-project-id.iam.gserviceaccount.com"
+}
+```
+
+Generally, you have to create a Service Account and assign it the following roles or the permissions described on each role:
+
+* `Compute Viewer` (1 permission in use of 138):
+- `monitoring.metricDescriptors.list`
+- `monitoring.timeSeries.list`
+
+* `Monitoring Viewer` (2 permissions in use of 13)
+- `compute.instances.get`
+
+You can play in IAM pretty much with your service accounts and Instance level access to your resources (for example, allowing that everything running in an Instance is authorized to use the Compute API). The module uses Google Cloud Platform libraries for authentication so many possibilities are open but the Module is only supported by using the method mentioned above.
+
+[float]
+== Google Cloud Platform module: Under the hood
+
+Google Cloud Platform offers the https://cloud.google.com/monitoring/api/metrics_gcp[Stackdriver Monitoring API] to fetch metrics from its services. *Those metrics are retrieved one by one*.
+
+If you also want to *extract service metadata and labels* (by setting `exclude_labels` to false, which is the default state). You also make a new API check on the corresponding service. Service labels requires a new API call to extract those metrics. In the worst case the number of API calls will be doubled. In the best case, all metrics come from the same GCP entity and 100% of the required information is included in the first API call (which is cached for subsequent calls).
+
+[float]
+=== Rough estimation of the number of API Calls
+For example, imagine that you have a Compute Metricset activated and you don't want to exclude labels. You have a total of 20 instances running in a particular GCP project, region and zone.
+
+Compute Metricset fetches 14 metrics. Each of those metrics will attempt an API call to Compute API to retrieve also their metadata. Because you have 20 different instances, the total number of API calls that will be done on each refresh period are: 14 metrics + 20 instances = 34 API requests every 5 minutes if that is your current Period. 9792 API requests per day with one zone. If you add 2 zones more with the same amount of instances you'll have 19584 API requests per day (9792 on each zone) or around 587520 per month for the Compute Metricset. This maths must be done for each different Metricset with slight variations.
```

Comment:
The number of metrics fetched by a metricset can change with future improvements, could we rephrase this so we don't have to remember to update this if the metricset changes?

Reply:
Rephrased. I want to leave the numbers as they are but I have added a small explanation that "14 are the number of metrics fetched in the early release"